java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java

```diff
@@ -0,0 +1,82 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class OptionalTask1 {
+
+    private static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public static int[] inputElem() throws IOException {
+        System.out.println("Enter the number of elements:");
+        int numberOfElements = Integer.parseInt(reader.readLine());
+        int[] array = new int[numberOfElements];
+        System.out.println("Enter the elements:");
+        for (int i = 0; i < numberOfElements; i++) {
+            array[i] = Integer.parseInt(reader.readLine());
+        }
+        return array;
+    }
+
+    public static void findShortElement(int[] array) {
+        int shortElem = array[0];
+        for (int i = 1; i < array.length; i++) {
+            if (array[i] < shortElem) {
+                shortElem = array[i];
+            }
+        }
+        System.out.println("The shortest element is " + shortElem);
+        System.out.println("The length of this element is " + elemLength(shortElem));
+    }
+
+    public static void findLongElement(int[] array) {
+        int longElem = array[0];
+        for (int i = 1; i < array.length; i++) {
+            if (array[i] > longElem) {
+                longElem = array[i];
+            }
+        }
+        System.out.println("The longest element is " + longElem);
+        System.out.println("The length of this element is " + elemLength(longElem));
+    }
+
+    public static void sort(int[] array) {
```

Comment:
rename to sortByLength to emphasize what function does

Reply:
done